stabilizer/src/main/java/com/hazelcast/stabilizer/provisioner/AwsProvisioner.java

```diff
@@ -0,0 +1,314 @@
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.stabilizer.common.AgentAddress;
+import com.hazelcast.stabilizer.common.AgentsFile;
+import com.hazelcast.stabilizer.common.StabilizerProperties;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+import static com.hazelcast.stabilizer.Utils.appendText;
+import static com.hazelcast.stabilizer.Utils.sleepMillis;
+
+
+/*
+* An aws specific provisioning class which is using the AWS sdk to create aws instances,  and aws elastic load balance
+*
+*
+* */
+public class AwsProvisioner {
+
+    private final static ILogger log = Logger.getLogger(AwsProvisioner.class);
+
+    //AWS specific magic strings
+    public static final String AWS_RUNNING_STATE  = "running";
+    public static final String AWS_SYSTEM_STATUS_OK = "ok";
+    public static final String AWS_PUBLIC_IP_FILTER = "ip-address";
+
+
+    //the file wich will hole the public domain name of the created load balance
+    public static final String AWS_ELB_FILE_NAME = "aws-elb.txt";
+
+    private static final int SLEEPING_MS = 1000 * 30;
+    private static final int MAX_SLEEPING_ITTERATIONS = 12;
+
+
+    private AWSCredentials credentials;
+    private AmazonEC2 ec2;
+    private AmazonElasticLoadBalancingClient elb;
+
+    private StabilizerProperties props = new StabilizerProperties();
+
+
+    private final File agentsFile = new File("agents.txt");
+    private final File elbFile = new File(AWS_ELB_FILE_NAME);
+
+    private File credentialsFile;
+
+    private String securityGroup;
+
+    private String awsKeyName;
+    private String awsAmi;
+    private String awsBoxId;
+
+    private String elbProtocol;
+    private int elbPortIn;
+    private int elbPortOut;
+    private String elbAzs;
+
+
+    public AwsProvisioner() throws IOException {
+        props.init(null);
+
+        String awsCredentialsPath = props.get("AWS_CREDENTIALS", "AwsCredentials.properties");
+        credentialsFile = new File(awsCredentialsPath);
+
+        elbProtocol = props.get("ELB_PROTOCOL");
+
+        elbPortIn = Integer.parseInt(props.get("ELB_PORT_IN"));
+        elbPortOut = Integer.parseInt(props.get("ELB_PORT_OUT"));
+        elbAzs = props.get("ELB_AZS");
+
+        awsKeyName = props.get("AWS_KEY_NAME");
+        awsAmi = props.get("AWS_AMI");
+        awsBoxId = props.get("AWS_BOXID");
+        securityGroup = props.get("SECURITY_GROUP");
+
+
+        credentials = new PropertiesCredentials(credentialsFile);
+
+        ec2 = new AmazonEC2Client(credentials);
+        elb = new AmazonElasticLoadBalancingClient(credentials);
+    }
+
+
+
+    public void addAgentsToLoadBalancer(String elbName){
+
+        if( ! isBalancerAlive(elbName) ){
+            createLoadBalancer(elbName);
+        }
+
+        List<Instance> instances = getInstancesFromAgentsFile();
```

Comment:
every instance that runs the "Test"

tom@class=com.hazelcast.stabilizer.tests.webContainer.TomCatContainer
tom@webAppPath=webapp
tom@port=6555

will start a Tomcat server